### ISSUE TITLE
ASP .NET Core Unauthorised User After UserManager.CreateAsync

### ISSUE THREAD

**Issue by skeeler88** [MarimerLLC/cslaforum#829]
**Describe the bug**
Not exactly sure if this is a bug or an intended override.

The type Microsoft.AspNetCore.Identity.UserManager after injecting it into a Razor View and calling UserManager.CreateAsync(new ApplicationUser()) results in a unauthenticated for presently logged in user.

E.g. I create a CSLA .NET BO IsSavable is True then call UserManager.CreateAsync(...) then IsSavable on the object is False and Csla.ApplicationContext.User is unathenticated and the HttpContext.User is presently authenticated.

I found my workaround for present environment using this code... hasn't been thoroughly thought through but is enough for unit tests to pass.

```
public class ApplicationContextManager : Csla.Core.ApplicationContextManager
    {
        private IHttpContextAccessor HttpContextAccessor;

        public ApplicationContextManager(IHttpContextAccessor httpContextAccessor) : base()
        {
            HttpContextAccessor = httpContextAccessor;
        }

        public override IPrincipal GetUser()
        {
                            if (HttpContextAccessor.HttpContext.User.Identity.IsAuthenticated)
            {
                SetUser(HttpContextAccessor.HttpContext.User);
                return HttpContextAccessor.HttpContext.User;
            }

            return base.GetUser();
        }
    }
```

**Version and Platform**
CSLA version: 5.0.1
OS: Windows, Linux, iOS, Android, etc.
Platform: ASP .NET Core 2.2 & .NET Standard 2.0

**Code that Fails**
https://github.com/MarimerLLC/csla/blob/master/Source/Csla.Shared/Core/ApplicationContextManager.cs

`    /// <summary>
    /// Gets the current user principal.
    /// </summary>
    /// <returns>The current user principal</returns>
    public virtual IPrincipal GetUser()
    {
      IPrincipal result = Thread.CurrentPrincipal;
      if (result == null)
      {
        result = new Csla.Security.UnauthenticatedPrincipal();
        SetUser(result);
      }
      return result;
    }
`

**Stack Trace or Exception Detail**

1. Create a CSLA .NET Business Object via DataPortal.
2. Call UserManager.CreateAsync(...)
3. Try to save the CSLA .NET Business Object
4. Csla.SecurityException is thrown

**Comment by rockfordlhotka** [comment]
I think this is related to an issue I'm working on right now: https://github.com/MarimerLLC/csla/issues/1369

There's a bug right now where the `WebContextManager` isn't properly set for an aspnetcore app. Figuring out exactly how/when/where to set this value is a little tricky, but I think I figured it out today, where I can set it in `UseCsla` and capture the necessary serviceprovider reference to access all the things necessary (mostly the httpcontext accessor).

**Comment by rockfordlhotka** [comment]
I just pushed a prerelease of v5.1.0 that should fix the issue.

**Comment by skeeler88** [comment]
Fixed in 5.1
